c7n/registry.py

```diff
@@ -60,12 +60,14 @@ def subscribe(self, event, func):
         self._subscribers[event].append(func)
 
     def register(self, name, klass=None, condition=True,
-                 condition_message="Missing dependency for {}"):
+                 condition_message="Missing dependency for {}",
+                 aliases=None):
         if not condition and klass:
             return klass
         # invoked as function
         if klass:
             klass.type = name
+            klass.aliases = aliases
```

Comment:
can we prefix this (perhaps as type_aliases)? type attribute stomping on registered classes is long standing, if we're introducing a new stamped attribute, it would be good to try and ensure its not stomping on an implementation attribute.

Reply:
renamed and updated validation tests for the first two resources using aliases.